test/e2e/e2e.go

```diff
@@ -2210,6 +2275,141 @@ var _ = ginkgo.Describe("e2e ingress traffic validation", func() {
+	var endPoints []*v1.Pod
+	var nodesHostnames sets.String
+	maxTries := 0
+	var nodes *v1.NodeList
+	var externalIpv4 string
+	var externalIpv6 string
+
+	// This test validates ingress traffic to nodeports with externalTrafficPolicy Set to local.
+	// It creates a nodeport service on both udp and tcp, and creates a host networked
+	// backend pod on each node. The backend pod is using the agnhost - netexec command which
+	// replies to commands with different protocols. We use the "hostname" and "clientip" commands
+	// to have each backend pod to reply with its hostname and the request packet's srcIP.
+	// We use an external container to poke the service exposed on the node and ensure that only the
+	// nodeport on the node with the backend actually receives traffic and that the packet is not
+	// SNATed.
+	ginkgo.Context("Validating ingress traffic to Host Netwoked pods", func() {
+		ginkgo.BeforeEach(func() {
+			endPoints = make([]*v1.Pod, 0)
+			nodesHostnames = sets.NewString()
+
+			var err error
+			nodes, err = e2enode.GetBoundedReadySchedulableNodes(f.ClientSet, 3)
+			framework.ExpectNoError(err)
+
+			if len(nodes.Items) < 3 {
+				framework.Failf(
+					"Test requires >= 3 Ready nodes, but there are only %v nodes",
+					len(nodes.Items))
+			}
+
+			ginkgo.By("Creating the endpoints pod, one for each worker")
+			for _, node := range nodes.Items {
+				// this create a udp / http netexec listener which is able to receive the "hostname"
+				// command. We use this to validate that each endpoint is received at least once
+				args := []string{
+					"netexec",
+					fmt.Sprintf("--http-port=%d", endpointHTTPPort),
+					fmt.Sprintf("--udp-port=%d", endpointUDPPort),
+				}
+
+				// create hostNeworkedPods
+				hostNetPod, err := createPod(f, node.Name+"-hostnet-ep", node.Name, f.Namespace.Name, []string{}, hostNetEndpointsSelector, func(p *v1.Pod) {
+					p.Spec.Containers[0].Args = args
+					p.Spec.HostNetwork = true
+				})
+
+				framework.ExpectNoError(err)
+				endPoints = append(endPoints, hostNetPod)
+				nodesHostnames.Insert(hostNetPod.Name)
+
+				// this is arbitrary and mutuated from k8s network e2e tests. We aim to hit all the endpoints at least once
+				maxTries = len(endPoints)*len(endPoints) + 30
+			}
+
+			ginkgo.By("Creating an external container to send the traffic from")
+			// the client uses the netexec command from the agnhost image, which is able to receive commands for poking other
+			// addresses.
+			externalIpv4, externalIpv6 = createClusterExternalContainer(clientContainerName, agnhostImage, []string{"--network", "kind", "-P"}, []string{"netexec", "--http-port=80"})
+		})
+
+		ginkgo.AfterEach(func() {
+			deleteClusterExternalContainer(clientContainerName)
+			// Since we're using host neworked pods in the test wait until namespaces delete after each test
+			framework.WaitForNamespacesDeleted(f.ClientSet, []string{f.Namespace.Name}, wait.ForeverTestTimeout)
```

Comment:
is this needed? the framework should clean everything in the namespace :thinking:

Reply:
So with host networked pods I wan't to ensure things are cleaned up before continuing